SnappyAPITests/APIWithStubsTests.swift

```diff
@@ -0,0 +1,118 @@
+    func testGetAllImagesNotValidJSON() {
+        
+        let expectationGetAllImages = expectation(description: "getting all images with not valid json")
+        
+        OHHTTPStubs.stubRequests(passingTest: { (request) -> Bool in
+            return true
+        }) { (request) -> OHHTTPStubsResponse in
+            let stubData = "NOT VALID JSON".data(using: .utf8)
+            return OHHTTPStubsResponse(data: stubData!, statusCode: 200, headers: nil)
+        }
+        
+        SnapchatAPI.getImages { (response) in
+            switch response {
+            case .success(_):
+                XCTAssert(false)
+            case .failure(_):
+              XCTAssert(true)
+            }
+            
+            expectationGetAllImages.fulfill()
+        }
+
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+    
+    
+    
+    func testAUploadImageWith400Error() {
+        let exp = expectation(description: "upload image with 400 error")
+        
+        OHHTTPStubs.stubRequests(passingTest: { (request) -> Bool in
+            return true
+        }) { (request) -> OHHTTPStubsResponse in
+            let stubData = "{\"error\": \"Stub is working!\"}".data(using: .utf8)
+            return OHHTTPStubsResponse(data: stubData!, statusCode: 400, headers: nil)
+        }
+        
+        let bundle = Bundle(for: self.classForCoder)
+        let image = UIImage(contentsOfFile: bundle.path(forResource: "thunder", ofType: "png")!)
+        
+        if let image = image {
+            SnapchatAPI.upload(image: image, completion: { response in
+                switch response {
+                case .failure(let error):
+                    exp.fulfill()
+                    XCTAssertEqual(error.localizedDescription, "Stub is working!")
+                default:
+                break
+                }
+            })
+        } else {
+            XCTAssert(false)
+        }
+        
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+    
+    
+    func testAUploadImageWithNoError() {
+        let exp = expectation(description: "upload image with no error")
+        
+        OHHTTPStubs.stubRequests(passingTest: { (request) -> Bool in
+            return true
+        }) { (request) -> OHHTTPStubsResponse in
+            let stubData = "{\"Success\": \"Image uploaded correctly.\"}".data(using: .utf8)
+            return OHHTTPStubsResponse(data: stubData!, statusCode: 400, headers: nil)
+        }
+        
+        let bundle = Bundle(for: self.classForCoder)
+        let image = UIImage(contentsOfFile: bundle.path(forResource: "thunder", ofType: "png")!)
+        
+        if let image = image {
+            SnapchatAPI.upload(image: image, completion: { response in
+                switch response {
+                case .success(_):
+                     exp.fulfill()
+                    XCTAssertEqual(response.value.debugDescription, "Optional({\n    Success = \"Image uploaded correctly.\";\n})")
```

Comment:
should it be before `fulfill`? Also, this assert seems really weird with this text:
```
Optional({\n    Success = \"Image uploaded correctly.\";\n})
```

maybe we can find another way to check it? parse to json dict and check both "Success" keys?